### PR TITLE
Biundo/graphql mapped types

### PR DESCRIPTION
**Pull request by johnbiundo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: content/graphql/mapped-types.md [hunk 1]
```diff
@@ -1,14 +1,16 @@
 ### Mapped types
 
-##### This chapter applies only to code first approach
+> warning **Warning** This chapter applies only to the code first approach.
 
-A common task is to take an existing type and make each of its properties optional. This happens often enough that NestJS provides a way to create new types based on old types — mapped types.
+As you build out features like CRUD (Create/Read/Update/Delete) it's often useful to construct variants on a base entity type. Nest provides several utility functions that perform type transformations to make this task more convenient.
 
-#### Partial
+#### Partial: PartialType() function
```
Comment (kamilmysliwiec) [inline content/graphql/mapped-types.md:7]: I'm not sure, but I feel like `PartialType()` looks somewhat weird as part of the heading

File: content/graphql/mapped-types.md [hunk 2]
```diff
@@ -64,11 +69,9 @@ export class UpdateEmailInput extends PickType(CreateUserInput, ['email']) {}
 
 > info **Hint** The `PickType()` function is imported from the `@nestjs/graphql` package.
 
-#### Pick
-
-Constructs a type by picking all properties from type and then removing particular set of keys.
+#### Omit: OmitType() function
```
Comment (kamilmysliwiec) [inline content/graphql/mapped-types.md:72]: Same as above

File: content/graphql/mapped-types.md [hunk 3]
```diff
@@ -33,13 +35,16 @@ export class UpdateUserInput extends PartialType(CreateUserInput) {}
...
+The `PartialType()` function takes an optional second argument that is a reference to the decorator factory of the type being extended. In the example above, we are extending `CreateUserInput` which is annotated with the `@InputType()` decorator. We didn't need to pass `InputType` as the second argument since it's the default value. If you want to extend a class decorated with `@ObjectType`, pass `ObjectType` as the second argument. For example:
 
-#### Pick
+```typescript
+@InputType()
+export class UpdateUserInput extends PartialType(User, ObjectType) {}
+```
 
-Constructs a type by picking the set of properties from type.
+#### Pick: PickType() function
```
Comment (kamilmysliwiec) [inline content/graphql/mapped-types.md:45]: Same as above

File: content/graphql/resolvers-map.md [hunk 4]
```diff
@@ -308,19 +308,23 @@ type Query {
...
-Args:
+You can use standard TypeScript class inheritance to create base classes with generic utility type features (fields and field properties, validations, etc.) that can be extended. For example, you may have a set of pagination related arguments that always include the standard `offset` and `limit` fields, but also other index fields that are type-specific. You can set up a class hierarchy as shown below.
+
+Base `@ArgsType()` class:
 
 ```typescript
 @ArgsType()
 class PaginationArgs {
-  @Field(type => Int)
+  @Field((type) => Int)
```
Comment (kamilmysliwiec) [inline content/graphql/resolvers-map.md:318]: For the sake of consistency, should we remove parenthesis? `(type)` -> `type` (see other examples above)

File: content/graphql/mapped-types.md (hunk 1 above)
Comment (johnbiundo) [reply content/graphql/mapped-types.md:7]: Yeah, I had mixed feelings too.  Figured I'd put it out there and see how you reacted :)  I'll come up with something better...

File: content/graphql/resolvers-map.md (hunk 4 above)
Comment (johnbiundo) [reply content/graphql/resolvers-map.md:318]: Yes.  Do you prefer to keep it as (type => Int) vs. () => Int?  I prefer the second myself, as I was initially confused about whether some argument is passed, but I'm OK either way.

**Comment by johnbiundo** [comment]
@kamilmysliwiec 
- most changes discussed were made
- I did *not* add inheritance example in mutations chapter.  Just removed your comment for now. Wasn't confident enough I had a good example, and decided we probably want to get this update completed ASAP.  Feel free to disagree and add an example if you like, and I can make another pass.
- I did *not* move the CLI section to its own page.  Maybe I can tackle that after this PR.

File: content/graphql/resolvers-map.md (hunk 4 above)
Comment (kamilmysliwiec) [reply content/graphql/resolvers-map.md]: TypeORM convention is `type => Int`. Perhaps, we should stick to this approach(?) (to be coherent)

**Comment by kamilmysliwiec** [comment]
Thank you @johnbiundo!

**Pull request merged**